Responses API: 500 error in certain streaming tool call responses

### System Info

Running from latest main as of this report - commit f3284368312858bd6d9a42cd3608ceaf712f8455

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### 🐛 Describe the bug

When testing the Codex CLI locally with a smaller model (RedHatAI/Llama-3.2-3B-Instruct-FP8) and the remote-vllm provider, I'm getting a 500 error from my Llama Stack server.

My test setup:

```
vllm serve RedHatAI/Llama-3.2-3B-Instruct-FP8 \
  --max-model-len 32000 \
  --enable-auto-tool-choice \
  --tool-call-parser llama3_json
```

```
INFERENCE_MODEL="RedHatAI/Llama-3.2-3B-Instruct-FP8" \
llama stack run llama_stack/templates/remote-vllm/run.yaml
```

```
export OPENAI_BASE_URL="http://meldjotne.local:8321/v1/openai/v1"                                    
export OPENAI_API_KEY="fake"
cd ~/src/llama-stack
codex --model "RedHatAI/Llama-3.2-3B-Instruct-FP8" \
  "Summarize the recent changes to this git repo.()"
```

It looks like we're not handling the case where the arguments value of the tool call returned from the inference server can be `None` initially in the streaming tool call and later that gets turned into an actual string.

### Error logs

```
 19:23:38.775 [ERROR] Error in sse_generator
Traceback (most recent call last):
  File "/Volumes/SourceCode/llama-stack/llama_stack/distribution/server/server.py", line 180, in sse_generator
    async for item in event_gen:
  File "/Volumes/SourceCode/llama-stack/llama_stack/providers/inline/agents/meta_reference/openai_responses.py", line 495, in _create_streaming_response
    response_tool_call.function.arguments += tool_call.function.arguments
TypeError: unsupported operand type(s) for +=: 'NoneType' and 'str'
```

### Expected behavior

The response should stream back without error.

--- a/llama_stack/providers/inline/agents/meta_reference/openai_responses.py
+++ b/llama_stack/providers/inline/agents/meta_reference/openai_responses.py
@@ -492,7 +492,12 @@ class OpenAIResponsesImpl:
                     for tool_call in chunk_choice.delta.tool_calls:
                         response_tool_call = chat_response_tool_calls.get(tool_call.index, None)
                         if response_tool_call:
-                            response_tool_call.function.arguments += tool_call.function.arguments
+                            # Don't attempt to concatenate arguments if we don't have any new arguments
+                            if tool_call.function.arguments:
+                                # Guard against an initial None argument before we concatenate
+                                response_tool_call.function.arguments = (
+                                    response_tool_call.function.arguments or ""
+                                ) + tool_call.function.arguments
                         else:
                             tool_call_dict: dict[str, Any] = tool_call.model_dump()
                             tool_call_dict.pop("type", None)

